### PR TITLE
Remove backward-compatibility hacks for old configuration system

### PR DESCRIPTION
**Pull request by astrofrog**
This addresses https://github.com/astropy/astropy/issues/2767

* [x] fix tests
* [x] changelog entry
* [x] send a warning email to astropy-dev to warn that the configuration items in affiliated packages will need to be updated in order to be compatible with Astropy 1.2

The package template didn't actually use the old configuration system (it didn't have any example configuration items at all). So https://github.com/astropy/package-template/pull/57 doesn't actually have to be merged yet (or at all), so we can focus on finishing up this PR first, and then we can consider separately whether to include an example in the package template.

### CODE REVIEW

File: astropy/vo/client/vos_catalog.py [hunk 1]
```diff
@@ -543,7 +539,7 @@ def from_registry(cls, registry_url, timeout=60, **kwargs):
 
         """
         # Download registry as VO table
-        with REMOTE_TIMEOUT.set_temp(timeout):
+        with conf.remote_timeout.set_temp(timeout):
```
Comment (pllim) [inline astropy/vo/client/vos_catalog.py]: This is either incorrect or confusing. This `conf` is most likely the one imported in Line 20, which is from `io.votable`, not `utils.data`. Also at some other places in the same module, `conf` is overwritten by `from .. import conf`. I think the `conf` import lines need some clean up.

Comment (astrofrog) [inline astropy/vo/client/vos_catalog.py]: Yes, I've fixed this in an upcoming commit

Comment (pllim) [inline astropy/vo/client/vos_catalog.py]: Okay, thanks! :+1:

**Comment by embray** [comment]
Haven't looked at the code changes yet but :+1: from me in principle.

**Comment by astrofrog** [comment]
I sent a message to astropy-dev about this and gave a deadline of Monday to merge this if all tests pass.

**Comment by bsipocz** [comment]
@astrofrog - This (and the other PRs opened today) will close #3500, too.

**Comment by astrofrog** [comment]
The tests are passing, so this is now ready for review.

**Comment by embray** [comment]
PRs that remove an order of magnitude more code than they add are the best.  Looks good to me.

**Comment by eteq** [comment]
:+1: from me!

**Comment by astrofrog** [comment]
Had to rebase due to a changelog entry conflict, but will then merge.

**Comment by astrofrog** [comment]
There have been no objections raised on astropy-dev. Merging!

**Comment by embray** [comment]
I like to remove "Ready-for-final-review" after it's been closed.  Though it mostly doesn't matter.

**Pull request merged**